.githooks/pre-push

```diff
@@ -0,0 +1,43 @@
+#! /usr/bin/env ruby
+
+$stderr.puts("Running pre-push validation...")
+
+lines = `git fetch origin main 2> /dev/null && git diff --name-only origin/main | grep -E "\.rbi$|index.json"`.lines
+files = lines.map(&:strip).select { |file| File.file?(file) }
+
+if lines.empty?
+  $stderr.puts("Nothing to check")
+  $stderr.puts("\n")
+  exit(0)
+end
+
+success = true
+
+$stderr.puts("Linting RBI files...")
+unless system("bundle exec rubocop #{files.join(" ")}")
+  success = false
+end
+$stderr.puts("Type checking RBI files...")
+unless system("scripts/check_types #{files.join(" ")}")
+  success = false
+end
+$stderr.puts("Checking that new RBI files belong to public gems...")
+unless system("scripts/check_gems_are_public #{files.join(" ")}")
+  success = false
+end
+$stderr.puts("Linting index.json...")
+unless system("scripts/lint_index_json")
+  success = false
+end
+$stderr.puts("Checking that index.json entries match rbi/annotations/*.rbi files...")
+unless system("scripts/match_index_entries_with_rbis")
+  success = false
+end
```

Comment:
I added a `"test": {}` entry in the index file and somehow this didn't catch any problem.

Reply:
Ok, it works, not sure why I didn't get the error the first time 👍

Reply:
It successfully stopped push for me as well 👍